From f83e8de1b9f832f9d3ef54ea4ba7f5694ac5a3ed Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Wed, 3 Apr 2013 15:21:51 +0200
Subject: [PATCH 247/364] 	* grub-core/term/i386/pc/console.c
 (grub_console_getwh): Decrease 	reported width by one to compensate
 for curesor algorithm problem.

---
 ChangeLog                        | 5 +++++
 grub-core/term/i386/pc/console.c | 3 ++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 0592cac..b0c57bb 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,10 @@
 2013-04-03  Vladimir Serbinenko  <phcoder@gmail.com>
 
+	* grub-core/term/i386/pc/console.c (grub_console_getwh): Decrease
+	reported width by one to compensate for curesor algorithm problem.
+
+2013-04-03  Vladimir Serbinenko  <phcoder@gmail.com>
+
 	Fix screen corruption in menu entry editor and simplify the code
 	flow while on it.
 
diff --git a/grub-core/term/i386/pc/console.c b/grub-core/term/i386/pc/console.c
index 2aa1943..ee6650b 100644
--- a/grub-core/term/i386/pc/console.c
+++ b/grub-core/term/i386/pc/console.c
@@ -255,7 +255,8 @@ grub_console_getkeystatus (struct grub_term_input *term __attribute__ ((unused))
 static grub_uint16_t
 grub_console_getwh (struct grub_term_output *term __attribute__ ((unused)))
 {
-  return (80 << 8) | 25;
+  /* Due to current cursor moving algorithm we lost the last column.  */
+  return (79 << 8) | 25;
 }
 
 static void
-- 
1.8.1.4

